gamelogic.js

```diff
@@ -1,13 +1,67 @@
 function displayWordSoFar(word, guesses) {
-  // WRITE ME
+  //code to display "_ _ _ _" in accordance to the length of the word
+  // console.log(word);
+  // console.log(guesses);
+
+  // let array = [];
+  //
+  // for (let i = 0; i < word.length; i++) {
+  //   array.push('_ ');
+  //
+  // }
+  //
+  // let text = array.join('').toString();
+  // return text;
+
+  //code to display where the letters that are guessed so far appear on it
+  let output = "";
+  for( let i = 0; i < word.length; i++) {
+    if (guesses.includes(word[i])) {
+      output = output + word[i] + " ";
+    } else {
+      output = output + "_ ";
+    }
+  }
+  return output;
 }
 
 function isGameWon(word, guesses) {
-  // WRITE ME
+  let output = displayWordSoFar(word, guesses);
+  output = output.split(" ").join("");
+  console.log(output);
+
+  if (word === output) {
+    return true;
+  } else {
+    return false;
+  }
+  // return word === output;
 }
 
 function isGameLost(word, guesses) {
-  // WRITE ME
+  //maak een array voor foute letters in op te slaan
+  let wrongLetters = [];
+  // guesses = [];
+  for (let i = 0; i < guesses.length; i++) {
+    if (guesses[i] !== word) {
+      wrongLetters.push(guesses[i]);
```

Comment:
Nee, dat gaat niet goed. In word staat het woord 'javascript', in guesses[i] staat een geraden letter.
Al kun je het opschrijven het is letters met woorden vergelijken, of appels met peren.
Gewoon de include functie gebruiken, net als in de andere twee functies. En omdat je tot 7 moest tellen, vond ikzelf het handiger om gewoon een tellertje te gebruiken. Het kan natuurlijk wel met push en de lengte nemen, maar het heeft geen toegevoegde waarde.

Ik had hem zo ( ik had het tekenen van het galgje hier ook in gezet):

```
function isGameLost(word, guesses) {
    // WRITE ME
    let wrongCount = 0;

    // loop over the guesses array
    for (let i = 0; i < guesses.length; i++) {
        // if a character in guesses is not in word, add 1 to wrongCount
        if (!word.includes(guesses[i])) {
            wrongCount++;
            //if wrongCount >= 7 game is lost
            if (wrongCount >= 7) {
                //displayGallow(wrongCount);
                return true;
            }
        }
    }
   // displayGallow(wrongCount);

    // if we get here, game is not lost (yet)
    return false;
}

```